pgmpy/estimators/EM.py

```diff
@@ -113,13 +103,22 @@ def _get_log_likelihood(self, datapoint: Dict[str, Any]) -> float:
         likelihood = 0
         for cpd in self.model_copy.cpds:
             scope = set(cpd.scope())
+            if any(pd.isna(datapoint.get(var)) for var in scope):
+                continue
             likelihood += log(
                 max(
                     cpd.get_value(
                         **{
-                            key: value
+                            key: (
+                                int(value)
+                                if not pd.isna(value)
+                                and isinstance(
+                                    value, (int, float, np.integer, np.floating)
+                                )
+                                else value
+                            )
```

Comment:
Why is this change required? Doesn't the logic in `_parallel_compute_weights` already make sure that there are no Nan values before the `_get_log_likelihood` call?